src/components/common/YourCompany/ChangeBusinessType.vue

```diff
@@ -433,24 +433,20 @@ export default class ChangeBusinessType extends Mixins(CommonMixin) {
   submitTypeChange () {
     this.setEntityType(this.selectedEntityType)
     this.isEditingType = false
-    if (this.shouldUpdateName()) {
```

Comment:
We would only update the name and set the NR if the business name had **NOT** changed. This prevents the case when a user attempts to change the business type (and therefore the name) more than once, as the business name **HAS** changed.

Reply:
Does this logic work on a named company? With and without a new NR?

Reply:
Since these changes still retain the check for a numbered business, as before, named companies should not be reaching this block of code below. There should be separate logic that handles named businesses (I can verify where this is if you like).

Reply:
> I did a little test on a named BC Limited, and could not change the type to ULC without a NR.

I thought this was the expected behaviour for named businesses? Since altering the business type for a named business would require a name change and this requires an NR.

Reply:
Yes, but what if wanted to change my named BC to a numbered ULC? Anyway, we discussed this previously and decided it was low priority right now.